packages/transaction-controller/src/TransactionController.ts

```diff
@@ -378,7 +378,7 @@ export class TransactionController extends BaseController<
       status: TransactionStatus.confirmed,
       time,
       transaction: {
-        chainId: 1,
+        chainId: parseInt(currentChainId, undefined),
```

Comment:
It can be hex or decimal, but that is resolved in the next PR (currently up as a draft). This pattern is unfortunate but temporary and following existing conventions for this module